// @aws-sdk/client-cloudwatch/dist/es/commands/EnableAlarmActionsCommand.js
import { __extends } from "tslib";
import { EnableAlarmActionsInput } from "../models/models_0";
import { deserializeAws_queryEnableAlarmActionsCommand, serializeAws_queryEnableAlarmActionsCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Enables the actions for the specified alarms.</p>
 */
var EnableAlarmActionsCommand = /** @class */ (function (_super) {
    __extends(EnableAlarmActionsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function EnableAlarmActionsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    EnableAlarmActionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "EnableAlarmActionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: EnableAlarmActionsInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    EnableAlarmActionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryEnableAlarmActionsCommand(input, context);
    };
    EnableAlarmActionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryEnableAlarmActionsCommand(output, context);
    };
    return EnableAlarmActionsCommand;
}($Command));
export { EnableAlarmActionsCommand };
